src/nunavut/lang/c/templates/definitions.j2

```diff
@@ -46,7 +46,18 @@ static_assert({{ t | full_reference_name }}_EXTENT_BYTES_ >= {# -#}
 {%- endfor %}
 {% for f in t.fields_except_padding if f.data_type is ArrayType %}
 /// Array metadata for: {{ f }}
+{%- if options.enable_override_variable_array_capacity and f.data_type is VariableLengthArrayType %}
+#ifndef {{ t | full_reference_name }}_{{ f.name }}_ARRAY_CAPACITY_
+{%- endif %}
 #define {{ t | full_reference_name }}_{{ f.name }}_ARRAY_CAPACITY_           {{ f.data_type.capacity }}U
+{%- if options.enable_override_variable_array_capacity and f.data_type is VariableLengthArrayType %}
+#elif !defined(reg_drone_service_battery_Status_0_2_DISABLE_SERIALIZATION_BUFFER_CHECK)
+#  define reg_drone_service_battery_Status_0_2_DISABLE_SERIALIZATION_BUFFER_CHECK
```

Comment:
```suggestion
#elif !defined(reg_drone_service_battery_Status_0_2_DISABLE_SERIALIZATION_BUFFER_CHECK_)
#  define reg_drone_service_battery_Status_0_2_DISABLE_SERIALIZATION_BUFFER_CHECK_
```

By convention documented in the header comment, we add a single underscore at the end to reduce the chance of conflict with DSDL entities (what would happen if you have a constant named `DISABLE_SERIALIZATION_BUFFER_CHECK` in your DSDL)

Reply:
I'm having trouble with this change: why do we have `reg_drone_service_battery_Status_0_2_` anything in the templates? This is a single datatype in one UAVCAN profile. We can't pollute these templates with specific knowledge.

Reply:
The objective is to disable the check for this type only, keeping this on for other types. This is supposed to be safer than disabling the check for all generated code completely.